src/equisolve/numpy/scripts/equi_kit.py

```diff
@@ -0,0 +1,299 @@
+            #)])
+            #self.estimator = Ridge(parameter_keys=parameter_keys, alpha=empty_tm)
+
+    def fit(self, X: Tuple[TensorMap, ...], y: TensorMap, **kwargs) -> TEquiKitScript:
+        # X : (X0, X1, ..., XN)
+        self._set_and_check_fitting_parameters()
+        if "transformer_X" not in kwargs.keys():
+            kwargs["transformer_X"] = {}
+        if "transformer_y" not in kwargs.keys():
+            kwargs["transformer_y"] = {}
+        if "estimator" not in kwargs.keys():
+            kwargs["estimator"] = {}
+
+        if self.transformer_X is None:
+            self._transformers_X = None
+        else:
+            self._transformers_X = []
+            for i in range(len(T)):
+                self._transformers_X.append( deepcopy(transformer_X).fit(X[i], **kwargs["transformer_X"]) )
+
+        if self.transformer_y is None:
+            self._transformer_y = None
+        else:
+            self._transformer_y = deepcopy(self.transformer_y).fit(y, **kwargs["transformer_y"])
+
+        X = tuple(self._transformers_X[i].transform(X[i]) for i in range(len(self._transformers_X)))
+        y = self._transformer_y.transform(y)
+
+        X = self.transform_join(X)
+
+        self._estimator = deepcopy(self.estimator).fit(X, y, **kwargs["estimator"])
+
+    def forward(self, X: Tuple[TensorMap, ...]) -> TensorMap:
+        # TODO check if is fitted
+
+        if self._transformers_X is not None:
+            X = tuple(self._transformers_X[i].transform(X[i]) for i in range(len(self._transformers_X)))
+        if self._transformer_y is not None:
+            y = self._transformer_y.transform(y)
+
+        X = self.join(X)
+
+        y_pred = self._estimator.predict(X, y)
+        if self._transformer_y is not None:
+            y_pred = self._transformer_y.inverse_transform(y_pred)
+        return y_pred
+
+    def score(self, X: Tuple[TensorMap, ...], y) -> List[float]:
+        # TODO(low-prio) add support for more error functions
+        y_pred = self.predict(X, y)
+        return rmse(y, y_pred, self._parameter_keys)
+
+    def _set_and_check_fitting_parameters(self) -> None:
+        # TODO check if parameter_keys are consistent over the transformers and estimators
+        #      and if not throw warning
+
+        if self.feature_aggregation not in ["sum", "meam"]:
+            raise ValueError("Only 'sum' and 'mean' are supported for feature_aggregation")
+
+        # TODO would rename to _fit_*
+        # we save all member variables, to have the member variables used in the last fit
+        self._hypers = self.hypers
+        self._feature_aggregation = self._feature_aggregation
+        self._parameter_keys = self.parameter_keys
+
+    @abstractmethod
+    def join(self, X: Tuple[TensorMap, ...]) -> TensorMap:
+        # TODO make error message sound nicer, double check if Metaclass is basically useless and does not do this
+        raise NotImplemented("join function not implemented")
```

Comment:
This might do the job. 

```suggestion
        raise NotImplemented("Only implemented in child classes")
```